### PR TITLE
Initial additions for minScale and maxScale annotations

### LINKED ISSUE #1331: No docs on minScale/maxScale features
Knative.dev has no documentation on how to use minScale/maxScale stuff.

There's some info buried in autoscale-go sample https://knative.dev/docs/serving/samples/autoscale-go/ which I wouldn't normally read as a user trying to learn Knative.

There should be ideally a dedicated concept or how-to page explaining these.

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #1331 

## Proposed Changes

- added config details for minScale
- added config details for maxScale

### CODE REVIEW

File: docs/serving/configuring-the-autoscaler.md [hunk 1]
```diff
@@ -108,21 +108,43 @@ limit how many requests reach the app at a given time. Using
...
 
 If there is no `/target` annotation, the autoscaler is configured as if
 `/target` == `containerConcurrency`.
 
+### Configuring scale bounds (minScale and maxScale)
+
+The `minScale` and  `maxScale` annotations can be used to configure the minimum and maximum number of pods that can serve applications.
+These annotations can be used to prevent cold starts or to help control computing costs.
+
+`minScale` and `maxScale` can be configured as follows in the revision template;
```
Comment (markusthoemmes) [inline docs/serving/configuring-the-autoscaler.md]: Maybe we should stress **revision template** a bit more and make clear in the example that this is part of the template, i.e.

```yaml
spec:
  template:
    metadata:
      autoscaling.knative.dev/minScale: "2"
      autoscaling.knative.dev/maxScale: "10"
```

It's a common mistake to throw these on the top level object instead.

File: docs/serving/configuring-the-autoscaler.md [hunk 2]
```diff
@@ -108,21 +108,43 @@ limit how many requests reach the app at a given time. Using
...
+
+```
+autoscaling.knative.dev/minScale: "2"
+autoscaling.knative.dev/maxScale: "10"
+```
+
+Using these annotations in the revision template will propagate this to `PodAutoscaler` objects. You can also manually propagate `PodAutoscaler` objects.
```
Comment (markusthoemmes) [inline docs/serving/configuring-the-autoscaler.md]: Not sure what's meant by "manually propagate" here but we could state that `PodAutoscaler` objects are mutable and you can change these annotations after the fact on the respective `PodAutoscaler` object.

File: docs/serving/configuring-the-autoscaler.md [hunk 3]
```diff
@@ -108,21 +108,43 @@ limit how many requests reach the app at a given time. Using
...
+
+**NOTE:** These annotations apply for the full lifetime of a revision. Even when a revision is not referenced by any route, the minimal pod count specified by `minScale` will still be provided. Keep in mind that non-routeable revisions may be garbage collected, which enables Knative to reclaim the resources.
+
+#### Default behavior
+
+If the `minScale` annotation is not set, pods will scale to zero.
```
Comment (markusthoemmes) [inline docs/serving/configuring-the-autoscaler.md]:
```suggestion
If the `minScale` annotation is not set, pods will scale to zero (or to 1 if `enable-scale-to-zero` is `false` per the ConfigMap mentioned above).
```

File: docs/serving/configuring-the-autoscaler.md (hunk 1 above)
Comment (markusthoemmes) [reply docs/serving/configuring-the-autoscaler.md]: Second thoughts: That's likely true for the entire document, so could be fixed in a separate PR.

File: docs/serving/configuring-the-autoscaler.md (hunk 3 above)
Comment (abrennan89) [reply docs/serving/configuring-the-autoscaler.md]: @markusthoemmes @samodell apparently we need author consent to commit Markus's change? Not sure how to do this. Should I remove it somehow and commit it myself or is it OK to add this way?

Comment (markusthoemmes) [reply docs/serving/configuring-the-autoscaler.md]: @abrennan89 feel free to rebase the PR and thus remove my name from the commit entirely.

File: docs/serving/configuring-the-autoscaler.md [hunk 4]
```diff
@@ -108,38 +109,70 @@ limit how many requests reach the app at a given time. Using
...
+    metadata:
+      autoscaling.knative.dev/metric: concurrency
+      autoscaling.knative.dev/class: hpa.autoscaling.knative.dev
 ```
 
 ## Additional resources
 
 - [Go autoscaling sample](https://knative.dev/docs/serving/samples/autoscale-go/index.html)
-- [Knative v0.3 Autoscaling  - A Love Story blog post](https://medium.com/knative/knative-v0-3-autoscaling-a-love-story-d6954279a67a)
+- ["Knative v0.3 Autoscaling  - A Love Story" blog post](https://medium.com/knative/knative-v0-3-autoscaling-a-love-story-d6954279a67a)
```
Comment (samodell) [inline docs/serving/configuring-the-autoscaler.md]: Medium blogs are now on Knative.dev -- https://knative.dev/blog/2019/03/27/knative-v0.3-autoscaling-a-love-story/

**Comment by samodell** [comment]
One small tweak; then we can get this merged in, @abrennan89 !

**Comment by samodell** [comment]
Thanks @abrennan89 !

/lgtm
/approve

**Pull request merged**